tokenizers/src/models/unigram/trainer.rs

```diff
@@ -421,6 +424,11 @@ impl UnigramTrainer {
             Vec::with_capacity(self.vocab_size.try_into().unwrap());
         // XXX: Make sure unk exists and are ids 0
         pieces.push((self.unk_token.clone(), f64::NAN));
+        pieces.extend(
+            self.special_tokens
+                .iter()
+                .map(|tok| (tok.content.clone(), f64::NAN)),
```

Comment:
I would feel more confident if we had a test for this NaN behaviour.

It's supposed to be taken care of upward in the stack I know, but I don't think there's actually a test to train on content that contains that kind of token, *and* that the actual tokenization of the resulting tokenizer does handle this special tokens correctly.